service-proxy.js

```diff
@@ -353,6 +354,18 @@ function removeServicePeer(serviceName, hostPort) {
     }
 
     if (!anyOtherSubChan) {
+        var connections = peer.connections;
+        var allDrained = CountedReadySignal(connections.length);
+        allDrained(onAllDrained);
+        for (var j = 0; j < connections.length; j++) {
+            connections[j].drain('closing due to unadvertisement', allDrained.signal);
+        }
+    }
+
+    function onAllDrained() {
+        self.logger.info('Peer drained and closed', self.extendLogInfo({
```

Comment:
should use peer.extendLogInfo, but that doesn't exist yet... and now the repos are separated, so we have to add it first before we use it here...